solr/core/src/test/org/apache/solr/handler/component/SearchHandlerTest.java

```diff
@@ -283,6 +289,82 @@ public void testRequireZkConnectedDistrib() throws Exception {
+
+    final CloudSolrClient cloudSolrClient = miniCluster.getSolrClient();
+
+    try {
+      assertNotNull(miniCluster.getZkServer());
+      List<JettySolrRunner> jettys = miniCluster.getJettySolrRunners();
+      assertEquals(1, jettys.size());
+      for (JettySolrRunner jetty : jettys) {
+        assertTrue(jetty.isRunning());
+      }
+
+      // create collection
+      String collectionName = "testLuceneIOExceptionHandling";
+      String configName = collectionName + "Config";
+      miniCluster.uploadConfigSet(
+          SolrTestCaseJ4.TEST_PATH().resolve("collection1/conf"), configName);
+
+      CollectionAdminRequest.createCollection(collectionName, configName, 2, 1)
+          .setPerReplicaState(SolrCloudTestCase.USE_PER_REPLICA_STATE)
+          .process(miniCluster.getSolrClient());
+
+      for (int i = 0; i < 5; i++) {
+        SolrInputDocument doc = new SolrInputDocument();
+        doc.addField("id", i);
+        doc.addField("title", "test" + i);
+        cloudSolrClient.add(collectionName, doc);
+      }
+      cloudSolrClient.commit(collectionName);
+
+      Collection<BytesRef> terms = new ArrayList<>();
+      for (int i = 0; i < 10; i++) {
+        terms.add(newBytesRef("term" + i));
+      }
+      TermInSetQuery termInSetQuery = new TermInSetQuery("name", terms);
+      SolrQuery solrQuery = new SolrQuery(termInSetQuery.toString());
+      final QueryRequest req = new QueryRequest(solrQuery);
+      req.setMethod(SolrRequest.METHOD.POST);
+      BaseHttpSolrClient.RemoteSolrException e =
+          assertThrows(
+              BaseHttpSolrClient.RemoteSolrException.class,
+              () -> req.process(cloudSolrClient, collectionName));
+      assertEquals(400, e.code());
+      assertTrue(e.getMessage().contains("org.apache.solr.search.SyntaxError"));
+
+      solrQuery = new SolrQuery("{!surround maxBasicQueries=1 df=title}99W(test1,test2,test3)");
+      final QueryRequest req2 = new QueryRequest(solrQuery);
+      req2.setMethod(SolrRequest.METHOD.POST);
+      e =
+          assertThrows(
+              BaseHttpSolrClient.RemoteSolrException.class,
+              () -> req2.process(cloudSolrClient, collectionName));
+      assertEquals(400, e.code());
+      assertTrue(e.getMessage().contains("Exceeded maximum of 1 basic queries."));
+
+      solrQuery = new SolrQuery("{!surround maxBasicQueries=1000 df=title}10W(tes*,test*)");
+      final QueryRequest req3 = new QueryRequest(solrQuery);
+      req3.setMethod(SolrRequest.METHOD.POST);
+      e =
+          assertThrows(
+              BaseHttpSolrClient.RemoteSolrException.class,
+              () -> req3.process(cloudSolrClient, collectionName));
+      assertEquals(400, e.code());
+      assertTrue(
+          e.getMessage()
+              .contains("Query contains too many nested clauses; maxClauseCount is set to 1"));
+    } finally {
+      System.clearProperty("solr.max.booleanClauses");
```

Comment:
Is there a chance that this might have been set to some value before we set it within this test? And if so, would we want to capture the value of this param and restore it or clear it conditionally here?

I'm honestly not sure, and my best guess is that this is fine as-is. May be a more important question if/when it comes to upstreaming this.